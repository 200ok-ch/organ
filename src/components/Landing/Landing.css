.landing-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.landing-container a {
    text-decoration: none;
}

.landing-button {
    text-align: center;

    font-size: 20px;
    margin-top: 30px;
    width: 200px;
}

.landing-app-name {
    font-size: 80px;
    font-weight: 300;

    margin-top: 0;
    margin-bottom: 20px;
}

.landing-tagline {
    font-size: 24px;
    font-weight: 300;

    margin-top: 0;
    margin-bottom: 5px;
}

.view-sample-button {
    margin-top: 40px;
}
