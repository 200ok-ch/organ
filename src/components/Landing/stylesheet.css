.main-image {
  max-height: 35em;
}
.testimonial-brand img {
  height: 3em;
  width: auto;
}

/* Icons with <img> */
#icons img {
  opacity: 0.5;
  margin-bottom: 2em;
  width: 100%;
  max-height: 2em;
}

#icons img:hover {
  opacity: 1;
}

/* Icons with Fontawesome <i> */
#icons svg:not(:root).svg-inline--fa {
  height: 3em;
  width: 100%;
}

#false-bottom-preventor {
  cursor: pointer;
}

.youtube-video-container {
  position: relative;
  overflow: hidden;
  width: 100%;
}

.youtube-video-container::after {
  display: block;
  content: '';
  padding-top: 56.25%;
}

.youtube-video-container iframe {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
