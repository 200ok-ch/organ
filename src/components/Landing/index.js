import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import './stylesheet.css';
import logo from '../../images/organice.svg';
import ExternalLink from '../UI/ExternalLink';
import * as baseActions from '../../actions/base';

const Landing = ({ base }) => {
  return (
    <main className="landing-container-wrapper">
      <div className="landing-container">
        <h1 className="landing-app-name">organice</h1>

        <img className="landing-logo" src={logo} alt="Logo" />

        <h2 className="landing-tagline">organice organizes Org files nicely!</h2>

        <h2 className="landing-tagline">
          Syncs with Dropbox,
          <br /> Google Drive and WebDAV.
        </h2>

        <p className="landing-description">
          organice allows you to view and edit Org files from cloud storage directly on your device!
          No Org file or other user-data will be stored on our servers; the entire app is
          browser-based.
        </p>

        <Link to="/sign_in">
          <div className="btn landing-button">Sign in</div>
        </Link>
        <Link to="/sample" onClick={() => base.restoreStaticFile('sample')}>
          <div className="btn landing-button doc-button">View sample</div>
        </Link>
        <ExternalLink href="https://organice.200ok.ch/documentation.html">
          <div className="btn landing-button doc-button">Documentation</div>
        </ExternalLink>
      </div>
      <footer>
        <Link to="/privacy-policy">Privacy Policy</Link>
      </footer>
    </main>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    base: bindActionCreators(baseActions, dispatch),
  };
};

export default connect(null, mapDispatchToProps)(Landing);
