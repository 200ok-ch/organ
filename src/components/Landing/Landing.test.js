import React from 'react';
import Landing from '../Landing';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

test('<Landing /> renders', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Landing />
    </MemoryRouter>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('<Landing /> handles sign in click', () => {
  const handleSignInClick = jest.fn();

  const component = mount(
    <MemoryRouter>
      <Landing onSignInClick={handleSignInClick} />
    </MemoryRouter>
  );

  const signInButton = component.findWhere(node => {
    try {
      return node.text().trim() === 'Sign in';
    } catch(error) {
      return false;
    }
  });

  signInButton.simulate('click');

  expect(handleSignInClick).toHaveBeenCalled();
});
