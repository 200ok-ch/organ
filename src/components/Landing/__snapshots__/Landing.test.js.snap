// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> renders 1`] = `
<div>
  <main>
    <div
      class="landing-container"
    >
      <h1
        class="landing-app-name"
      >
        organice
      </h1>
      <img
        alt="Logo"
        class="landing-logo"
        src="organice.svg"
      />
      <h2
        class="landing-tagline"
      >
        organice organizes Org files nicely!
      </h2>
      <h2
        class="landing-tagline"
      >
        Syncs with Dropbox,
        <br />
         Google Drive and WebDAV.
      </h2>
      <a
        href="/sample"
      >
        <div
          class="btn landing-button view-sample-button"
        >
          View sample
        </div>
      </a>
      <a
        href="/sign_in"
      >
        <div
          class="btn landing-button"
        >
          Sign in
        </div>
      </a>
    </div>
    <footer>
      <a
        href="/privacy-policy"
      >
        Privacy Policy
      </a>
    </footer>
  </main>
</div>
`;
